src/realizations/catchment/LSTM_Realization.cpp

```diff
@@ -10,55 +10,22 @@ using namespace realization;
  * Parameterized constructor for LSTM Realization with lstm_params and lstm_config
  * already constructed.
  *
- * @param forcing_config
- * @param output_stream
  * @param catchment_id
+ * @param forcing
+ * @param output_stream
  * @param params
  * @param config
  */
 LSTM_Realization::LSTM_Realization(
-        forcing_params forcing_config,
+        std::string id,
+        std::shared_ptr<data_access::GenericDataProvider> gdp,
         utils::StreamHandler output_stream,
-        std::string catchment_id,
         lstm::lstm_params params,
-        lstm::lstm_config config)
-    : Catchment_Formulation(catchment_id, forcing_config, output_stream),
-      catchment_id(catchment_id), params(params), config(config)
-{
-    model = make_unique<lstm::lstm_model>(lstm::lstm_model(config, params));
-}
+        lstm::lstm_config config
+    ) : Catchment_Formulation(id, gdp, output_stream), params(params), config(config) {
+        model = make_unique<lstm::lstm_model>(lstm::lstm_model(config, params));
+    }
 
-/**
- * Parameterized constructor for LSTM Realization with individual paths and parameters
- * needed lstm_params and lstm_config passed.
- *
- * @param forcing_config
- * @param output_stream
- * @param catchment_id
- * @param pytorch_model_path
- * @param normalization_path
- * @param initial_state_path
- * @param latitude
- * @param longitude
- * @param area_square_km
- */
-LSTM_Realization::LSTM_Realization(
```

Comment:
Also, any reason this constructor got removed?

Reply:
It was not used anywhere. It could be re-added with `forcing_params` changed to a `DataProvider`, but it would just be zombie code.

Reply:
The intent of it was to ensure that if someone wanted to use the realization for something they could create it directly with the paths (it is a convenience  constructor).  It wasn't exactly meant to be used in our own code.

Reply:
My guess it it was using a forcing config object, not sure why it was removed an not refactored though.

Reply:
Somewhat subjective, but it's unused in the application... this isn't a general purpose library (at least not yet)... so it's arguably goldplating. Just more code to maintain.

Reply:
Nuke it if you want...we have moved away in large part from the development kit style of implementing realizations, relying instead on BMI for the moment.  I would, however, suggest that that deprecation/removal be done in its own isolated commit so that a history/blame on the file in the future would show a clean way to revive that constructor.